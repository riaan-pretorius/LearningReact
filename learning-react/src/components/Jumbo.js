import React, { Component } from 'react';
import { Jumbotron, Button } from 'reactstrap';


class Jumbo extends Component {
    constructor(props) {
        super(props);
        this.state = {
            title: props.title,
            subText: props.subText,
            buttonText: props.buttonText
        };
    }
    render() {
        return (
            <Jumbotron style={styles.jumbotron}>
                <h1>{this.state.title}</h1>
                <p>{this.state.subText}</p>
                <Button color="success">{this.state.buttonText}</Button>
            </Jumbotron>
        )
    }
}

const styles = {
    jumbotron: {
        textAlign: 'center'
    }
}

export default Jumbo