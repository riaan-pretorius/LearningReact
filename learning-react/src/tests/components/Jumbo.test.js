import React from 'react'
import Jumbo from '../../components/Jumbo'
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer'

test('Jumbo Test Snapshot', () => {
  const component = renderer.create(
    <Jumbo title="Jumbotron heading"
      subText="Cras justo odio, dapibus ac facilisis in, egestas eget quam. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus."
      buttonText="Sign up today" />
  )
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
})

test('Jumbo Test', () => {
    const component = mount(
        <Jumbo title="Jumbotron heading"
        subText="Cras justo odio."
        buttonText="Sign up today" />
    )
    expect(component.find('.jumbotron').exists()).toBeTruthy()

    let header = component.find('h1')
    expect(header.exists()).toBeTruthy()
    expect(header.text()).toBe("Jumbotron heading")

    let p = component.find('p')
    expect(p.exists()).toBeTruthy()
    expect(p.text()).toBe("Cras justo odio.")

    let button = component.find('button')
    expect(button.exists()).toBeTruthy()
    expect(button.text()).toBe("Sign up today")
})