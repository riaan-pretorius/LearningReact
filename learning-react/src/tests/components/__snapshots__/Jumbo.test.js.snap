// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jumbo Test Snapshot 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    Jumbotron heading
  </h1>
  <p>
    Cras justo odio, dapibus ac facilisis in, egestas eget quam. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus.
  </p>
  <button
    className="btn btn-success"
    onClick={[Function]}
    type={undefined}
  >
    Sign up today
  </button>
</div>
`;
